lib/spec/staticcacher.sw-spec.js

```diff
@@ -0,0 +1,181 @@
+
+describe('Static cacher', function() {
+
+  var defaultCacheName = 'offline';
+  var defaultUrls = ['http://www.example.com/', 'http://www.mozilla.org/'];
+  var defaultStaticCacher = new StaticCacher(defaultUrls);
+
+  sinon.stub(self, 'fetch', function() {
+    var response = new Response('content', {});
+    return Promise.resolve(response);
+  });
+
+  afterEach(function() {
+    self.fetch.reset();
+    return clearDefaultCache();
+  });
+
+  function clearDefaultCache() {
+    return caches.open(defaultCacheName).then(function(cache) {
+      return cache.keys().then(function(response) {
+        response.forEach(function(element) {
+          cache.delete(element);
+        });
+      });
+    });
+  }
+
+
+  describe('The constructor', function() {
+
+    it('should accept an array of files', function() {
+      var fileList = ['file1', 'file2'];
+      var cacher = new StaticCacher(fileList);
+      expect(cacher.files).to.deep.equal(fileList);
+    });
+
+    it('should throw an error if no argument is provided', function() {
+      expect(function() {
+        new StaticCacher();
+      }).to.throw(Error, 'Invalid file list');
+    });
+
+    it('should throw an error if the argument is not an array', function() {
+      expect(function() {
+        new StaticCacher('invalidString');
+      }).to.throw(Error, 'Invalid file list');
+    });
+
+    it('should throw if the array is empty', function() {
+      expect(function() {
+        new StaticCacher([]);
+      }).to.throw(Error, 'Invalid file list');
+    });
+  });
+
+  describe('onInstall()', function() {
+
+    it('should return a promise', function() {
+      expect(defaultStaticCacher.onInstall()).to.be.an.instanceOf(Promise);
+    });
+
+    it('should use the default cache', function() {
+      sinon.spy(defaultStaticCacher, 'getDefaultCache');
+
+      return defaultStaticCacher.onInstall().then(function() {
+        sinon.assert.calledOnce(defaultStaticCacher.getDefaultCache);
+      });
+    });
+
+    it('should call addAll files to the cache', function() {
+      sinon.spy(defaultStaticCacher, 'addAll');
+
+      return defaultStaticCacher.getDefaultCache().then(function(defaultCache) {
+        return defaultStaticCacher.onInstall().then(function() {
+          sinon.assert.calledWith(defaultStaticCacher.addAll, defaultCache, defaultUrls);
+        });
+      });
+    });
+  });
+
+  describe('getDefaultCache()', function() {
+    it('should return the offline cache', function() {
+      var cacheGot;
+
+      return defaultStaticCacher.getDefaultCache().then(function(cache) {
+        cacheGot = cache;
+        return caches.open(defaultCacheName)
+      }).then(function(expectedCache) {
+        expect(cacheGot).to.equal(expectedCache);
+      });
+    });
+  });
+
+  describe('addAll()', function() {
+    afterEach(function() {
+      defaultStaticCacher.fetchAndCache.restore();
+    });
+
+    it('should work with polyfill', function() {
```

Comment:
Maybe you would want to say `should work as polyfill`. I don't understand this very well, sorry.